### PR TITLE
[Tachyon-1267] Javadoc thrift definitions

### PR DESCRIPTION
**Pull request by ShaoshanLiu**
1. add javadoc to thrift definitions
2. address some merge conflicts

### CODE REVIEW

File: common/src/thrift/block_master.thrift [hunk 1]
```diff
@@ -15,51 +15,56 @@ struct WorkerInfo {
...
+  /**
    * Marks the given block as committed.
    */
-  void workerCommitBlock(1: i64 workerId, 2: i64 usedBytesOnTier, 3: string tierAlias, 4: i64 blockId,
-      5: i64 length)
+  void workerCommitBlock( /** the id of the worker */  1: i64 workerId,
+      /** the space used in bytes on the target tier */ 2: i64 usedBytesOnTier,
+      /** the alias of the target tier */ 3: string tierAlias,
+      /** the id of the block being committed */ 4: i64 blockId,
+      /** the id of the block being committed */ 5: i64 length)
```
Comment (jsimsa) [inline common/src/thrift/block_master.thrift]: the length

File: common/src/thrift/block_master.thrift [hunk 2]
```diff
@@ -15,51 +15,56 @@ struct WorkerInfo {
...
-  /*
-   * Periodic worker heartbeat.
+  /**
+   * Periodic worker heartbeat returns an optional command for the worker to execute
    */
-  common.Command workerHeartbeat(1: i64 workerId, 2: map<string, i64> usedBytesOnTiers,
-      3: list<i64> removedBlockIds, 4: map<string, list<i64>> addedBlocksOnTiers)
+  common.Command workerHeartbeat( /** the id of the worker */ 1: i64 workerId,
+      /** the map of space used in bytes on all tiers */ 2: map<string, i64> usedBytesOnTiers,
+      /** the list of removed block Ids */ 3: list<i64> removedBlockIds,
```
Comment (jsimsa) [inline common/src/thrift/block_master.thrift]: ids

File: common/src/thrift/file_master.thrift [hunk 3]
```diff
@@ -47,119 +47,125 @@ struct SetStateTOptions {
...
 
-  // Tachyon Client API
-
-  /*
+  /**
    * Marks a file as completed.
    */
-  void completeFile(1: i64 fileId, 2: CompleteFileTOptions options)
+  void completeFile( /** the id of the file */ 1: i64 fileId,
+       /** the options for completing the file */ 2: CompleteFileTOptions options)
```
Comment (jsimsa) [inline common/src/thrift/file_master.thrift]: I would just simply say "method options"

File: common/src/thrift/file_master.thrift [hunk 4]
```diff
@@ -47,119 +47,125 @@ struct SetStateTOptions {
...
     throws (1: exception.TachyonTException e)
 
-  /*
-   * Creates a file.
+  /**
+   * Creates a file and returns the id of the file
```
Comment (jsimsa) [inline common/src/thrift/file_master.thrift]: add a missing period at the end

File: common/src/thrift/file_master.thrift [hunk 5]
```diff
@@ -47,119 +47,125 @@ struct SetStateTOptions {
...
    */
-  i64 create(1: string path, 2: CreateTOptions options)
+  i64 create( /** the path of the file */ 1: string path,
+      /** the options for completing the file */ 2: CreateTOptions options)
     throws (1: exception.TachyonTException e, 2: exception.ThriftIOException ioe)
 
-  /*
-   * Frees the given file from Tachyon.
+  /**
+   * Frees the given file from Tachyon and returns whether the free operation succeeded
```
Comment (jsimsa) [inline common/src/thrift/file_master.thrift]: ditto

File: common/src/thrift/file_master.thrift [hunk 6]
```diff
@@ -47,119 +47,125 @@ struct SetStateTOptions {
...
-  i64 loadMetadata(1: string ufsPath, 2: bool recursive)
+  i64 loadMetadata( /** the path of the under file system */ 1: string ufsPath,
+      /** whether to load meta data recursively */ 2: bool recursive)
     throws (1: exception.TachyonTException e, 2: exception.ThriftIOException ioe)
 
-  /*
-   * Creates a directory.
+  /**
+   * Creates a directory and returns whether the directory is
+   * created successfully
```
Comment (jsimsa) [inline common/src/thrift/file_master.thrift]: add missing period and remove line break

File: common/src/thrift/file_master.thrift [hunk 7]
```diff
@@ -47,119 +47,125 @@ struct SetStateTOptions {
...
-  bool mount(1: string tachyonPath, 2: string ufsPath)
+  bool mount( /** the path of tachyon mount point */ 1: string tachyonPath,
+      /** the path of the under file system */ 2: string ufsPath)
     throws (1: exception.TachyonTException e, 2: exception.ThriftIOException ioe)
 
-  /*
-   * Deletes a file or a directory.
-   *
+  /**
+   * Deletes a file or a directory and returns whether the remove operation succeeded
```
Comment (jsimsa) [inline common/src/thrift/file_master.thrift]: add missing period

File: common/src/thrift/file_master.thrift [hunk 8]
```diff
@@ -47,119 +47,125 @@ struct SetStateTOptions {
...
    */
-  bool remove(1: i64 id, 2: bool recursive)
+  bool remove( /** the id of the file or directory */ 1: i64 id,
+      /** whether to remove recursively */ 2: bool recursive)
     throws (1: exception.TachyonTException e)
 
-  /*
-   * Renames a file or a directory.
+  /**
+   * Renames a file or a directory and returns whether the rename operation succeeded
```
Comment (jsimsa) [inline common/src/thrift/file_master.thrift]: ditto

File: common/src/thrift/lineage_master.thrift [hunk 9]
```diff
@@ -42,48 +42,52 @@ struct LineageInfo {
...
    * Marks a file as completed and initiates asynchronous persistence (if applicable).
    */
-  void asyncCompleteFile(1: i64 fileId)
+  void asyncCompleteFile( /** the id of the file */ 1: i64 fileId)
     throws (1: exception.TachyonTException e)
 
-  /*
-   * Creates a lineage.
+  /**
+   * Creates a lineage and returns the lineage id
```
Comment (jsimsa) [inline common/src/thrift/lineage_master.thrift]: add missing period

File: common/src/thrift/lineage_master.thrift [hunk 10]
```diff
@@ -42,48 +42,52 @@ struct LineageInfo {
...
+  /**
    * Returns a list of existing lineages.
    */
   list<LineageInfo> getLineageInfoList()
 
-  /*
-   * Reinitializes a file.
+  /**
+   * Reinitializes a file. Returns the id of the reinitialized file when the
+   * file is lost or not completed, -1 otherwise
```
Comment (jsimsa) [inline common/src/thrift/lineage_master.thrift]: ditto

File: common/src/thrift/lineage_master.thrift [hunk 11]
```diff
@@ -42,48 +42,52 @@ struct LineageInfo {
...
+  void reportLostFile( /** the path of the file */ 1: string path)
     throws (1: exception.TachyonTException e)
 
-  // Tachyon Worker API
-
-  /*
-   * Periodic lineage worker heartbeat.
+  /**
+   * Periodic lineage worker heartbeat. Returns the command for checkpointing
+   * the blocks of a file
```
Comment (jsimsa) [inline common/src/thrift/lineage_master.thrift]: ditto

File: common/src/thrift/raw_table_master.thrift [hunk 12]
```diff
@@ -14,36 +16,37 @@ struct RawTableInfo {
 
 service RawTableMasterService extends common.TachyonService {
 
-  // Tachyon Client API
-
-  /*
-   * Creates a raw table.
+  /**
+   * Creates a raw table and returns the raw table id
```
Comment (jsimsa) [inline common/src/thrift/raw_table_master.thrift]: ditto

File: common/src/thrift/worker.thrift [hunk 13]
```diff
@@ -4,9 +4,16 @@ include "common.thrift"
 include "exception.thrift"
 
 service WorkerService extends common.TachyonService {
-  void accessBlock(1: i64 blockId)
 
-  bool asyncCheckpoint(1: i64 fileId)
+  /**
+   * access a block given the block id
```
Comment (jsimsa) [inline common/src/thrift/worker.thrift]: `Accesses a block given the block id.`

File: common/src/thrift/worker.thrift [hunk 14]
```diff
@@ -4,9 +4,16 @@ include "common.thrift"
...
+   */
+  void accessBlock( /** the id of the block being accessed */ 1: i64 blockId)
+
+  /**
+   * asynchronously checkpoint a file: return whether the checkpoint operation succeeded
```
Comment (jsimsa) [inline common/src/thrift/worker.thrift]: `Asynchronously checkpoints a file: return whether the checkpoint operation succeeded.`

**Comment by jsimsa** [comment]
@ShaoshanLiu thanks for the improvement ... left some minor style comments

File: common/src/thrift/block_master.thrift (hunk 1 above)
Comment (ShaoshanLiu) [inline common/src/thrift/block_master.thrift]: fixed

File: common/src/thrift/block_master.thrift (hunk 2 above)
Comment (ShaoshanLiu) [inline common/src/thrift/block_master.thrift]: fixed

File: common/src/thrift/file_master.thrift (hunk 3 above)
Comment (ShaoshanLiu) [inline common/src/thrift/file_master.thrift]: fixed

File: common/src/thrift/file_master.thrift (hunk 4 above)
Comment (ShaoshanLiu) [inline common/src/thrift/file_master.thrift]: fixed

File: common/src/thrift/file_master.thrift (hunk 5 above)
Comment (ShaoshanLiu) [inline common/src/thrift/file_master.thrift]: fixed

File: common/src/thrift/file_master.thrift (hunk 6 above)
Comment (ShaoshanLiu) [inline common/src/thrift/file_master.thrift]: fixed

File: common/src/thrift/file_master.thrift (hunk 7 above)
Comment (ShaoshanLiu) [inline common/src/thrift/file_master.thrift]: fixed

File: common/src/thrift/file_master.thrift (hunk 8 above)
Comment (ShaoshanLiu) [inline common/src/thrift/file_master.thrift]: fixed

File: common/src/thrift/lineage_master.thrift (hunk 9 above)
Comment (ShaoshanLiu) [inline common/src/thrift/lineage_master.thrift]: fixed

File: common/src/thrift/lineage_master.thrift (hunk 10 above)
Comment (ShaoshanLiu) [inline common/src/thrift/lineage_master.thrift]: fixed

File: common/src/thrift/lineage_master.thrift (hunk 11 above)
Comment (ShaoshanLiu) [inline common/src/thrift/lineage_master.thrift]: fixed

File: common/src/thrift/raw_table_master.thrift (hunk 12 above)
Comment (ShaoshanLiu) [inline common/src/thrift/raw_table_master.thrift]: fixed

File: common/src/thrift/worker.thrift (hunk 13 above)
Comment (ShaoshanLiu) [inline common/src/thrift/worker.thrift]: fixed

File: common/src/thrift/worker.thrift (hunk 14 above)
Comment (ShaoshanLiu) [inline common/src/thrift/worker.thrift]: fixed

**Comment by ShaoshanLiu** [comment]
@jsimsa: please let me know if you have additional comments.  Thanks!

File: common/src/thrift/worker.thrift [hunk 15]
```diff
@@ -4,9 +4,16 @@ include "common.thrift"
 include "exception.thrift"
 
 service WorkerService extends common.TachyonService {
-  void accessBlock(1: i64 blockId)
 
-  bool asyncCheckpoint(1: i64 fileId)
+  /**
+   * access a block given the block id.
```
Comment (jsimsa) [inline common/src/thrift/worker.thrift]: `// Accesses ...`

File: common/src/thrift/worker.thrift [hunk 16]
```diff
@@ -4,9 +4,16 @@ include "common.thrift"
...
+   */
+  void accessBlock( /** the id of the block being accessed */ 1: i64 blockId)
+
+  /**
+   * asynchronously checkpoint a file: return whether the checkpoint operation succeeded.
```
Comment (jsimsa) [inline common/src/thrift/worker.thrift]: `Asynchronously checkpoints ...`

**Comment by jsimsa** [comment]
just two minor comments

**Comment by jsimsa** [comment]
also please resolve merge conflicts

**Comment by ShaoshanLiu** [comment]
@jsimsa do you know of a quick way to resolve merge conflicts instead of doing a rebase to replay each commit?

thanks!

**Comment by jsimsa** [comment]
what i do is that i set up the "upstream" origin in the local copy of my Tachyon repository fork:

```
$ git remote add upstream https://github.com/amplab/tachyon.git
```

Then, to merge the Tachyon master iinto the current branch, I run:

```
$ git fetch upstream
$ git merge upstream/master
```

Any conflicts that are not resolved automatically need to be resolved manually at which point you will have to run `git commit -a` to commit all manual conflict resolutions.

**Comment by ShaoshanLiu** [comment]
@jsimsa 
I am afraid this would bring in a lot of unrelated commits....

**Comment by jsimsa** [comment]
You mean conflicts with your PR? I don't think it will bring any more conflicts than rebase.

The unrelated commits won't be duplicated in the master branch history because you are doing a merge and not a rebase.

**Comment by ShaoshanLiu** [comment]
@jsimsa cool, thanks!

**Comment by ShaoshanLiu** [comment]
@jsimsa all review comments addressed, please let me know if you have additional comments.

thanks!

Shaoshan

**Comment by apc999** [comment]
+1 LGTM. Thanks for the improvement.

**Comment by jsimsa** [comment]
LGTM, thanks for the PR!

**Comment by jsimsa** [comment]
FYI, I am not sure what is the reason for the history of this PR to have most of your commits duplicated 4 times ... in the future, please try to keep your commit history clean

**Comment by ShaoshanLiu** [comment]
sure, mostly due to rebase and reapplying each commit when conflicts happened.  that's why i asked you how you resolved conflicts

On Dec 1, 2015 9:37 AM, Jiri Simsa <notifications@github.com> wrote:

FYI, I am not sure what is the reason for the history of this PR to have most of your commits duplicated 4 times ... in the future, please try to keep your commit history clean

-

**Comment by jsimsa** [comment]
one way to avoid that is to use merge instead of rebase (as per the instructions I shared with you)

**Pull request merged**